core/src/main/scala/org/broadinstitute/dsde/rawls/metrics/InstrumentationDirectives.scala

```diff
@@ -0,0 +1,18 @@
+package org.broadinstitute.dsde.rawls.metrics
+
+import spray.routing.Directive0
+import spray.routing.directives.BasicDirectives._
+import spray.routing.directives.MiscDirectives.requestInstance
+
+trait InstrumentationDirectives extends RawlsInstrumented {
+  def instrumentRequest: Directive0 = requestInstance flatMap { request =>
+    httpRequestCounter(request).inc()
+    mapRequestContext { context =>
+      val timeStamp = System.currentTimeMillis
+      context.withHttpResponseMapped { response =>
+        val rTime = httpRequestTimer(request).time(System.currentTimeMillis - timeStamp)
```

Comment:
It would be good to count the response (i.e. status code) here as well. Could we add a:
```
httpResponseCounter(response).inc()
```
?

Futuremore, maybe `httpResponseCounter` should take in the `HttpRequest` as well, so we can count (method, uri, response code) triples. We can modify `httpResponseCounter` to take in a `HttpRequest` like so:
```
httpResponseCounter(request, response).inc()
```